lib/methods/get-auth.js

```diff
@@ -1,36 +1,65 @@
 'use strict'
+const handleRejectPromise = (reject) => {
+  let err = new Error('No authentication found')
+  err.appWithoutAuth = true
+  reject(err)
+}
 
-const getAuth = ({ db, table }) => {
+const handleResolvePromise = (resolve, obj) => {
+  resolve({
+    row: obj,
+    // for Store API authentication headers
+    myId: obj.authentication_id,
+    accessToken: obj.access_token
+  })
+}
+const getAuth = ({ dbFilename, db, table }) => {
   return (storeId, authenticationId) => {
     return new Promise((resolve, reject) => {
       // select authentication for specified store from database
-      let query = 'SELECT * FROM ' + table + ' WHERE store_id = ? '
-      let params = [ storeId ]
-      if (authenticationId) {
-        // also filter by authentication ID
-        query += 'AND authentication_id = ? '
-        params.push(authenticationId)
-      }
-      // get one row only
-      query += 'LIMIT 1'
 
-      // run query and get row object
-      db.get(query, params, (err, row) => {
-        if (err) {
-          reject(err)
-        } else if (row) {
-          resolve({
-            row,
-            // for Store API authentication headers
-            myId: row.authentication_id,
-            accessToken: row.access_token
-          })
-        } else {
-          let err = new Error('No authentication found')
-          err.appWithoutAuth = true
-          reject(err)
+      //if have sqlite File
+      if(dbFilename){
+        let query = 'SELECT * FROM ' + table + ' WHERE store_id = ? '
+        let params = [ storeId ]
+        if (authenticationId) {
+          // also filter by authentication ID
+          query += 'AND authentication_id = ? '
+          params.push(authenticationId)
         }
-      })
+        // get one row only
+        query += 'LIMIT 1'
+
+        // run query and get row object
+        db.get(query, params, (err, row) => {
+          if (err) {
+            reject(err)
+          } else if (row) {
+            handleResolvePromise(resolve, row)
+          } else handleRejectPromise(reject)
+        })
+        //if dont have sqlite file, but have a firestore connection
+      }else{
+        let ref = db.collection(table)
+        let query
+        if(authenticationId){
+          query = ref.doc(authenticationId)
+        }else{
+          query = ref.where('store_id', '==', storeId).limit(1)
```

Comment:
Não deveria ser um `else` aqui, tem que bater o Store ID mesmo se o authentication ID for enviado (verifica os dois), assim como acontece com o SQLite..

Reply:
@leomp12 o .where é uma função do da collection. Se vc procura um documento usando o .doc(id) o where não funciona ! Como vc pediu pra fazer a busca pelo authenticationId no .doc essa foi a única solução que arrumei ! Porém já tenho pronta a solução para tirar o .doc e deixar tudo buscando como .where()